code/game/machinery/vending/vendor_types/squad_prep/squad_rifleman.dm

```diff
@@ -46,9 +46,21 @@ GLOBAL_LIST_INIT(cm_vending_clothing_marine, list(
 		list("Rebreather", 0, /obj/item/clothing/mask/rebreather, MARINE_CAN_BUY_MASK, VENDOR_ITEM_REGULAR),
 
 		list("RESTRICTED FIREARMS", 0, null, null, null),
-		list("VP78 Pistol", 15, /obj/item/storage/box/guncase/vp78, null, VENDOR_ITEM_REGULAR),
-		list("SU-6 Smart Pistol", 15, /obj/item/storage/box/guncase/smartpistol, null, VENDOR_ITEM_REGULAR),
-		list("M79 Grenade Launcher", 30, /obj/item/storage/box/guncase/m79, null, VENDOR_ITEM_REGULAR),
+
+		list("VP78", 0, null, null, null),
+		list("VP78 Pistol", 4, /obj/item/weapon/gun/pistol/vp78, null, VENDOR_ITEM_REGULAR),
+		list("VP78 Magazine", 2, /obj/item/ammo_magazine/pistol/vp78, null, VENDOR_ITEM_REGULAR),
+
+		list("SU-6", 0, null, null, null),
+		list("SU-6 Smart Pistol", 4, /obj/item/weapon/gun/pistol/smart, null, VENDOR_ITEM_REGULAR),
+		list("M276 pattern SU-6 Smartpistol holster rig", 1, /obj/item/storage/belt/gun/smartpistol, null, VENDOR_ITEM_REGULAR),//shouldn't be full.
+		list("SU-6 Smartpistol magazine (.45)", 2, /obj/item/ammo_magazine/pistol/smart, null, VENDOR_ITEM_REGULAR),
+
+		list("SU-6 Smart Pistol", 0, null, null, null),
```

Comment:
Is this meant to be SU-6 Smart Pistol? Probably meant to name it Grenade Launcher something right?